001-clarifications-eval.txt

```diff
@@ -0,0 +1,754 @@
+
+      * as the value of a debugging information entry attribute that is
+        encoded using class exprloc (see 7.5.5 Classes and Forms).
+
+      * as the operand to certain operation expression operations.
+
+      * as the operand to certain call frame information operations (see
+        6.4 Call Frame Information).
+
+      * in location list entries (see 2.5.x DWARF Location List
+        Expressions).
+    --------------------------------------------------------------------
+
+In Section 2.5.1.1 Literal Encodings, replace the description of
+DW_OP_const_type with the following:
+
+    --------------------------------------------------------------------
+    DW_OP_const_type has three operands. The first is an unsigned LEB128
+    integer DR that represents the byte offset of a debugging
+    information entry D relative to the beginning of the current
+    compilation unit, that provides the type T of the constant value.
+    The second is a 1-byte unsigned integral constant S. The third is a
+    block of bytes B, with a length equal to S.
+
+    TS is the bit size of the type T. The least significant TS bits of B
+    are interpreted as a value V of the type D. It pushes the value V
+    with the type D.
+
+    The DWARF is ill-formed if D is not a DW_TAG_base_type debugging
+    information entry in the current compilation unit, or if TS divided
+    by 8 (the byte size) and rounded up to a whole number is not equal
+    to S.
+
+    [non-normative] While the size of the byte block B can be inferred
+    from the type D definition, it is encoded explicitly into the
+    operation so that the operation can be parsed easily without
+    reference to the .debug_info section.
+    --------------------------------------------------------------------
+
+In Section 2.5.1.2 Register Values, replace the description of
+DW_OP_bregx with the following:
+
+    --------------------------------------------------------------------
+    DW_OP_bregx has two operands. The first is an unsigned LEB128
+    integer that represents a register number R. The second is a signed
+    LEB128 integer that represents a byte displacement B.
+
+    The action is the same as for DW_OP_breg<N>, except that R is used
+    as the register number and B is used as the byte displacement.
+    --------------------------------------------------------------------
+
+In Section 2.5.1.2 Register Values, replace the description of
+DW_OP_regval_type with the following:
+
+    --------------------------------------------------------------------
+    DW_OP_regval_type has two operands. The first is an unsigned LEB128
+    integer that represents a register number R. The second is an
+    unsigned LEB128 integer DR that represents the byte offset of a
+    debugging information entry D relative to the beginning of the
+    current compilation unit, that provides the type T of the register
+    value.
+
+    The operation is equivalent to performing DW_OP_regx R;
```

Comment:
I would argue that this is not (yet) allowed.

Reply:
I agree, but please take a look at my DW_OP_xderefX issue. I feel that we need to be consistent either way.

Reply:
Agreed and created issue #11 to move it to 004-locations-on-stack.txt .